Integrate measure builds plugin

wzieba opened the pull request:
## Description

This PR integrates [measure-builds-gradle-plugin](https://github.com/Automattic/measure-builds-gradle-plugin).


## To Test:

<!-- Test instructions per dependency update: https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/docs/test_instructions_per_dependency_update.md -->

### TC1 Without build scan

1. Make sure you updated secrets and you have `appsMetricsToken` value in `gradle.properties`.
2. Run any task with `--debug` flag, e.g. `./gradlew help --debug`
3. Find `REQUEST: https://metrics.a8c-ci.services/api/grouped-metrics` log

### TC2 With build scan

1. Make sure you updated secrets and you have `appsMetricsToken` value in `gradle.properties`.
2. Run any task with `--debug` flag and `CI=true` environmental variable, e.g. `CI=true ./gradlew help --debug`
3. Find `REQUEST: https://metrics.a8c-ci.services/api/grouped-metrics` log and assert, that in the sent payload, there's `{"name":"wordpress-gradle-scan-id","value":"<id of uploaded scan>"}` object.

## Demo 
<img width="797" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/5845095/0dc155f9-1451-4789-ae33-af3d89bc2bf7">

wzieba committed:
build: update secrets

wzieba committed:
build: integrate `measure-builds` plugin

wzieba commented:
hi @justtwago 👋 would you be ok reviewing this PR?

justtwago reviewed (approved):
Thank you a lot for integrating the plugin, @wzieba! Can't wait to see the first results and working on improving build times! 🎉 
Tested both cases and it works as described. ✅

wzieba committed:
build: bump `measure-builds` to `2.0.3`

The plugin no longer assumes that the authToken will be in `gradle.properties`, so we need to provide it in `measureBuilds` extension.

wzieba commented:
Thanks @justtwago for the review! Since it, I've added c5be91c - it only affects the authentication when sending metrics to the API. I've tested it and all works fine. Merging!
